### PR TITLE
cloud-sql-proxy v2.6.1 (new formula)

### PR DESCRIPTION
**Pull request by jhandguy**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/c/cloud-sql-proxy.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  license "Apache-2.0"
+
+  depends_on "go" => :build
+
+  def install
+    system "go", "build", *std_go_args(ldflags: "-s -w")
+  end
+
+  test do
+    assert_match "cloud-sql-proxy version #{version}", shell_output("#{bin}/cloud-sql-proxy --version")
```
[the remark this answers is not in the document]
Comment (jhandguy) [reply Formula/c/cloud-sql-proxy.rb:15]: > If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.

`cloud-sql-proxy` actually falls into this category - I added a test case to confirm that it fails due to lack of credentials.

File: Formula/c/cloud-sql-proxy.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    assert_match "could not find default credentials", shell_output("echo $(#{bin}/cloud-sql-proxy test 2>&1)")
```
Comment (SMillerDev) [inline Formula/c/cloud-sql-proxy.rb:16]:
```suggestion
    assert_match "could not find default credentials", shell_output("#{bin}/cloud-sql-proxy test 2>&1")
```

Comment (jhandguy) [reply Formula/c/cloud-sql-proxy.rb:16]: The `echo` here is used to suppress the exit 1 return status code (see other example in [kamel.rb](https://github.com/Homebrew/homebrew-core/blob/master/Formula/k/kamel.rb#L40)).
I found a better way to handle it using the second argument of `shell_output` though, thanks!

Comment (SMillerDev) [reply Formula/c/cloud-sql-proxy.rb]: Ah yeah, Kamel shouldn't do that either.

**Review by SMillerDev — APPROVED** [review]

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
@jhandguy thanks for your first PR to homebrew-core.

**Review by ZhongRuoyu — APPROVED** [review]
Thanks, @jhandguy!

**Pull request merged**